pareto/utilities/process_data.py

```diff
@@ -0,0 +1,932 @@
+                for s in strategic_model.s_S  # Storage Evaporation
+            )
+            + sum(
+                strategic_model.p_sigma_BeneficialReuse[o, t]
+                if strategic_model.p_sigma_BeneficialReuse[o, t].value >= 0
+                else strategic_model.p_M_Flow
+                for o in strategic_model.s_O  # Beneficial Reuse (if user does not specific value, p_sigma_BeneficialReuse is -1)
+                # In this case, Beneficial Reuse at this site has no limit (big M parameter)
+            )
+            + sum(
+                strategic_model.p_sigma_Storage[s]
+                + _get_max_value_for_parameter(strategic_model.p_delta_Storage)
+                for s in strategic_model.s_S  # Storage
+            )
+            + sum(
+                (
+                    strategic_model.p_sigma_Disposal[k]
+                    + _get_max_value_for_parameter(strategic_model.p_delta_Disposal)
+                )
+                * strategic_model.p_epsilon_DisposalOperatingCapacity[k, t]
+                for k in strategic_model.s_K  # (Initial Disposal + max disposal) * operating capacity
+            )
+            + sum(
+                max(
+                    [
+                        (
+                            strategic_model.p_sigma_Treatment[r, wt].value
+                            + max(
+                                [
+                                    strategic_model.p_delta_Treatment[wt, j].value
+                                    for j in strategic_model.s_J
+                                ]
+                            )
+                        )
+                        for wt in strategic_model.s_WT
+                    ]
+                )
+                for r in strategic_model.s_R
+            )  # Treatment: for each treatment site, select treatment technology that yields the
+            # maximum value for (initial treatment + max treatment expansion)
+        )
+        # Note: if completions pad is outside system, demand is not required to be met
+        strategic_model.p_gamma_TimePeriodDemand[t] = sum(
+            strategic_model.p_gamma_Completions[cp, t]
+            if strategic_model.p_chi_OutsideCompletionsPad[cp] == 0
+            else 0
+            for cp in strategic_model.s_CP
+        )
+
+        strategic_model.p_sigma_WaterAvailable[t] = strategic_model.p_beta_TotalPW[
+            t
+        ].value + sum(
+            strategic_model.p_sigma_ExternalWater[f, t] for f in strategic_model.s_F
+        )
+
+        # If volume of produced water is greater than capacity, raise an infeasibility error
+        if (
+            strategic_model.p_beta_TotalPW[t].value
+            > strategic_model.p_sigma_MaxPWCapacity[t].value
+        ):
+            capacity_feasibility_message.append(
+                f"{t} ({round(strategic_model.p_beta_TotalPW[t].value)} total {pyunits.get_units(strategic_model.p_beta_TotalPW[t])}s PW)"
+            )
```

Comment:
I think it would be a good idea to include the value of `p_sigma_MaxPWCapacity[t]` in the error message, so the user can easily see how bad the infeasibility is.

Reply:
Done!